indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java

```diff
@@ -97,13 +102,22 @@ public boolean isReady(TaskActionClient taskActionClient)
   @Override
   public void stopGracefully(TaskConfig taskConfig)
   {
+    aborted.set(true);
   }
 
   @Override
   public TaskStatus runTask(TaskToolbox toolbox) throws Exception
   {
-    Thread.sleep(runTime);
-    return TaskStatus.success(getId());
+    toolbox.getEmitter().emit(ServiceMetricEvent.builder().setMetric(NOOP_TASK_EVENT_STARTED, 1));
+    long endTime = System.currentTimeMillis() + runTime;
+    while (endTime > System.currentTimeMillis() && !aborted.get()) {
+      Thread.sleep(100);
+    }
```

Comment:
How about we use a `countDownLatch.await(runTime)` instead?
The latch will be counted down in `stopGracefully()`.
So we will wait on the latch until stop is requested or until the runTime elapses.